App.js

```diff
@@ -18,21 +25,38 @@ class App extends Component {
     }
 
     render() {
+        const {navigatorContainerStyle} = styles;
+
+        const MainNavigator = StackNavigator({
+            login: {
+                screen: LoginScreen
+            },
+            main: {
+                screen: TabNavigator({
+                    search: {screen: SearchScreen},
+                    favorites: {screen: FavoritesScreen},
+                    addSite: {screen: AddSiteScreen},
+                    more: {screen: MoreScreen}
+                })
+            },
+            filter: {
+                screen: FilterScreen,
+                navigationOptions: {
+                    tabBarVisible: false
+                }
+            }
+        }, {
+            lazy: true
+        });
+
         return (
             <Provider store={store}>
-                <Router/>
+                <MainNavigator/>
```

Comment:
The code above is MUCH cleaner than what we had to do in `Router.js` with `react-native-router-flux`.  The only issue is that I had a problem with nesting another `TabNavigator` for the `SearchScreen` so instead I had to manage it with state in that component. Not ideal, but it works.